### PR TITLE
KOGITO-3630: Create "EditExpressionMenu" component

### PR DESCRIPTION
**Pull request by vpellegrino**
**JIRA**: https://issues.redhat.com/browse/KOGITO-3630

_Preview of the component_
image
 _Brief architectural overview_
We are going to have several `Popover` instances in the editor, very similar to each other. So I created the `PopoverMenu` component as common base.
For instance, we will have the `SelectLogicTypeMenu` (in one of the next PRs) that will be used both in the `ExpressionContainer` and in every recursive Expression structure.
Regarding this PR, The `EditExpressionMenu` component will be used every time we need to have a popover with a "Name" to fill (input text) and a "Data type" to select (select box)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

File: kie-wb-common-react/boxed-expression-editor/src/api/ExpressionProps.ts
Comment (tiagobento) [inline kie-wb-common-react/boxed-expression-editor/src/api/ExpressionProps.ts]:
```suggestion
  expressionName?: string,
```

File: kie-wb-common-react/boxed-expression-editor/src/components/EditExpressionMenu/EditExpressionMenu.css
Comment (tiagobento) [inline kie-wb-common-react/boxed-expression-editor/src/components/EditExpressionMenu/EditExpressionMenu.css]: We should avoid CSS targeting element ids.. Using classes makes the components much more reusable.

**Review by tiagobento — COMMENTED** [review]
Hi @vpellegrino ! From what I can tell, you're going on the right direction. Minor comments inline, though.

The code is pretty much aligned with what we do on `kogito-tooling`, but I think you'd benefit a lot from using Prettier as the default formatter for these components too.

The tests are well-done using RTL and snapshots. Be ware that snapshot testing is not supposed to be the only way to perform assertions, and (for my taste), it's better to avoid having "call assertions". We should test what our users can see!

**Comment by vpellegrino** [comment]
> Hi @vpellegrino ! From what I can tell, you're going on the right direction. Minor comments inline, though.
> 
> The code is pretty much aligned with what we do on `kogito-tooling`, but I think you'd benefit a lot from using Prettier as the default formatter for these components too.
> 
> The tests are well-done using RTL and snapshots. Be ware that snapshot testing is not supposed to be the only way to perform assertions, and (for my taste), it's better to avoid having "call assertions". We should test what our users can see!

@tiagobento thanks for your useful comments and suggestions. I tried to accomplish all of them. Please let me know if something else needs to be changed.
_For the tests:_ About the "call assertions", IIUYC, you are referring to
```
    expect(mockedOnExpressionUpdate).toHaveBeenCalled();
    expect(mockedOnExpressionUpdate).toHaveBeenCalledWith({
      expressionName: "changed",
      dataType: DataType.Undefined,
    });
```
I just used them to test that changing the component status, the callback gets called, since the component is not yet included anywhere, otherwise I would have been tested the effects from a user point of view (as you are suggesting), for sure (and I will do as soon as it will be included).

Really appreciating your feedback and review.

**Comment by vpellegrino** [comment]
@tiagobento sorry for bothering you.
Waiting for your approval, before merging.
Thanks

**Comment by tiagobento** [comment]
@vpellegrino Sure! Glad we think alike in many aspects! Approving :)

**Review by tiagobento — APPROVED** [review]

**Pull request merged**
